packages/kbn-language-documentation-popover/src/utils/get_filtered_groups.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import type { LanguageDocumentationSections } from '../types';
+import { elementToString } from './element_to_string';
+
+export const getFilteredGroups = (
+  searchText: string,
+  searchInDescription?: boolean,
+  sections?: LanguageDocumentationSections,
+  numOfGroupsToOmit?: number
+) => {
+  const normalizedSearchText = searchText.trim().toLocaleLowerCase();
+  return sections?.groups
+    .slice(numOfGroupsToOmit ?? 0)
+    .map((group) => {
+      const options = group.items.filter((helpItem) => {
+        return (
+          !normalizedSearchText ||
+          helpItem.label.toLocaleLowerCase().includes(normalizedSearchText) ||
+          // Converting the JSX element to a string first
+          (searchInDescription &&
+            elementToString(helpItem.description)
+              ?.toLocaleLowerCase()
+              .includes(normalizedSearchText))
```

Comment:
Isn't this a bit too much work to do on every keystroke in the search? can the description be generated also in lowercase plain text just for that?

Reply:
This is not being intriduced in this PR but no we have decided that all our ESQL commands and functions will be in uppercase and it comes from ES .

Reply:
I'm not just talking about the lowercase operation (that, if that is the case, you can do the check by inverting to toLocaleUpperCase the searchText) but I'm talking about the `elementToString` that compute a plain text from a JSX everytime.

Reply:
Sorry then I am a bit confused. What are you suggesting?

Reply:
I'm suggesting to pre-compute only one time the `elementToString(helpItem.description)`, caching it or generating it, to avoid computing this everytime. At that point you can also precompute it in lowercase so you can skip also that passage

Reply:
Ok so we synced offline :D We concluded in an idea like this https://github.com/elastic/kibana/issues/192900 which I think is very cool

The comment about the lowercase I will do this in this PR as is very easy to do here.